docs/type.rst

```diff
@@ -46,54 +71,128 @@ Currently, supported basic types in Taichi are
     (OK: supported, EXT: require extension, N/A: not available)
 
 
-Boolean types should be represented using ``ti.i32``.
+.. note::
+
+    Boolean types are represented using ``ti.i32``.
+
 
-Binary operations on different types will give you a promoted type, following the C programming language, e.g.
 
-- ``i32 + f32 = f32``
-- ``f32 + f64 = f64``
-- ``i32 + i64 = i64``
+Type promotion
+--------------
+
+Binary operations on different types will give you a promoted type, following the C programming language convention, e.g.:
+
+- ``i32 + f32 = f32`` (integer + float = float)
+- ``i32 + i64 = i64`` (less-bits + more-bits = more-bits)
+
+Basically it will try to choose the more precise type to contain the result value.
 
 
 .. _default_precisions:
 
 Default precisions
 ------------------
 
-By default, numerical literals have 32-bit precisions.
+By default, all numerical literals have 32-bit precisions.
 For example, ``42`` has type ``ti.i32`` and ``3.14`` has type ``ti.f32``.
+
 Default integer and float-point precisions (``default_ip`` and ``default_fp``) can be specified when initializing Taichi:
 
 .. code-block:: python
 
-    ti.init(..., default_fp=ti.f32)
-    ti.init(..., default_fp=ti.f64)
+    ti.init(default_fp=ti.f32)
+    ti.init(default_fp=ti.f64)
+
+    ti.init(default_ip=ti.i32)
+    ti.init(default_ip=ti.i64)
+
+
+Also note that you may use ``float`` or ``int`` in type definitions as aliases
+for default precisions, e.g.:
+
+.. code-block:: python
+
+    ti.init(default_ip=ti.i64, default_fp=ti.f32)
+
+    x = ti.var(float, 5)
+    y = ti.var(int, 5)
+    # is equivalent to:
+    x = ti.var(ti.f32, 5)
+    y = ti.var(ti.i64, 5)
```

Comment:
```suggestion
    x = ti.field(dtype=float, shape=5)
    y = ti.field(dtype=int, shape=5)
    # is equivalent to:
    x = ti.field(dtype=ti.f32, shape=5)
    y = ti.field(dtype=ti.i64, shape=5)
```

Reply:
Thanks, let's only do `ti.var` -> `ti.field` after `rullec-field`, wdyt?

Reply:
OK, I think that's a good reason to stick to `ti.var` for now.